haystack/core/pipeline/pipeline.py

```diff
@@ -715,6 +716,13 @@ def run(self, word: str):
                 # Those are handled in a different way, so we skip them here.
                 continue
             instance = self.graph.nodes[component_name]["instance"]
+
+            # make sure that all components have all the inputs values explicitly defined even being optional
+            parameters = get_parameter_info(instance.run)
+            filled = {k: v["default_value"] for k, v in parameters.items() if k not in data and v["optional"]}
+            filled.update(component_inputs)
+            data[component_name] = filled
```

Comment:
I think we don't need to check the optionality at all in here. 🤔 
It should be enough to set a defaults if the Component has some.